package/MDAnalysis/lib/distances.py

```diff
@@ -303,6 +312,14 @@ def distance_array(reference: Union[npt.NDArray, 'AtomGroup'],
     distances = _check_result_array(result, (refnum, confnum))
     if len(distances) == 0:
         return distances
+    
```

Comment:
@orbeckst @richardjgowers we have an issue which is that the distopia backend doesn't seem to fill the result array on return. ie post call for the other backends, `result == distances` outside the function while for distopia its passed through. Not sure if I am doing something simple wrong. 

 Offending example in a test on this PR.
 
 ```python
     @pytest.mark.parametrize("dtype", (np.float32, np.float64))
    @pytest.mark.parametrize("backend", distopia_conditional_backend())
    def test_results_inplace_all_backends(self,  backend, dtype,):
        N = 10
        c0 = np.ones(3 * N, dtype=dtype).reshape(N, 3) * 2
        c1 = np.ones(3 * N, dtype=dtype).reshape(N, 3) * 3

        result = np.zeros(N, dtype=np.float64)
        distances.calc_bonds(c0, c1, result=result, backend=backend)
        expected = np.ones(N, dtype=dtype) * 3**(1/2)
        # test the result array is updated in place
        assert_almost_equal(result, expected, self.prec, err_msg="calc_bonds inplace failed")
 ```

We can get around this with a copy, but shouldn't be nessecary right? The distopia core API works inplace. Im fairly sure I am just getting some basic scoping wrong.

Reply:
This test shows distopia works in place. 

https://github.com/MDAnalysis/distopia/pull/179/files#diff-a0adc50f7cfca5f98b7c9829f3709cdfb652f5ed0325fffc37885bb1308b4f18R69

Reply:
I'd look into any use of [@check_coords](https://github.com/hmacdope/mdanalysis/blob/4fafd51de84d5b89be0559a412acefde0040847c/package/MDAnalysis/lib/util.py#L1965) as it creates _copies_ by default.

Reply:
(Maybe not relevant as you're concerned with results.)

Reply:
I think it's actually the `astype()` — see comment review.